packages/host/app/components/operator-mode/container.gts

```diff
@@ -587,9 +595,54 @@ export default class OperatorModeContainer extends Component<Signature> {
   };
 
   @action updateSubmode(submode: Submode) {
+    switch (submode) {
+      case Submode.Interact:
+        this.operatorModeStateService.updateCodePath(null);
+        break;
+      case Submode.Code:
+        let codePath = new URL(this.lastCardInRightMostStack.id);
+        this.operatorModeStateService.updateCodePath(codePath);
+        this.loadCardForCodeMode(codePath);
+        break;
+      default:
+        throw assertNever(submode);
+    }
+
     this.operatorModeStateService.updateSubmode(submode);
   }
 
+  @action resetCodeModeCardError() {
+    this.codeModeCardError = null;
+  }
+
+  @action loadCardForCodeMode(codePath: URL) {
+    this.loadCardForCodeModeTask.perform(codePath);
+  }
+
+  loadCardForCodeModeTask = restartableTask(async (codePath: URL) => {
```

Comment:
i think ultimately this will have to be refactored for loading a "file" in code mode, not a card--using the file resource that we currently use in our existing code route.

Reply:
Is there another mechanism we can use to find the realm? Perhaps based on just the URL?